src/qz/printer/status/CupsUtils.java

```diff
@@ -131,14 +123,15 @@ public static boolean clearSubscriptions() {
     static void startSubscription(int rssPort) {
         Runtime.getRuntime().addShutdownHook(new Thread(() -> freeIppObjs()));
 
-        Pointer request = cups.ippNewRequest(IPP.CREATE_PRINTER_SUBSCRIPTION);
+        Pointer request = cups.ippNewRequest(IPP.CREATE_JOB_SUBSCRIPTION);
 
         cups.ippAddString(request, IPP.TAG_OPERATION, IPP.TAG_URI, "printer-uri", CHARSET,
-                                   URIUtil.encodePath("ipp://localhost:" + IPP.PORT + "/printers"));
+                          URIUtil.encodePath("ipp://localhost:" + IPP.PORT + "/printers"));
         cups.ippAddString(request, IPP.TAG_OPERATION, IPP.TAG_NAME, "requesting-user-name", CHARSET, USER);
         cups.ippAddString(request, IPP.TAG_SUBSCRIPTION, IPP.TAG_URI, "notify-recipient-uri", CHARSET,
-                                   URIUtil.encodePath("rss://localhost:" + rssPort));
-        cups.ippAddString(request, IPP.TAG_SUBSCRIPTION, IPP.TAG_KEYWORD, "notify-events", CHARSET, "printer-state-changed");
+                          URIUtil.encodePath("rss://localhost:" + rssPort));
+        cups.ippAddStrings(request, IPP.TAG_SUBSCRIPTION, IPP.TAG_KEYWORD, "notify-events", 3, CHARSET,
```

Comment:
I think elements should be '2' here, not '3'.

Reply:
@Vzor- more context would be helpful.  I looked this up and found the following:
```cpp
ipp_attribute_t *ippAddStrings(ipp_t *ipp, ipp_tag_t group, ipp_tag_t value_tag, const char *name, int num_values, const char *language, const char *const *values);
```

Where:
* ipp_t *ipp = request (pointer)
* ipp_tag_t group = IPP_TAG_SUBSCRIPTION
* ipp_tag_t value_tag = "notify-events"
* name = 🚫 missing?
* int num_values = 3
* char *language = CHARSET
* char *values = {"job-state-changed", "printer-state-changed"}

So, I think what you're saying is we're passing two values ({"job-state-changed", "printer-state-changed"}) and therefor should have only specified 2 for num_values?   What about the missing `name` value?

Anyway, I think it's a copy/paste error from this example from [this markdown document](https://github.com/apple/cups/blob/master/cups/cupspm.md), notice it passes three elements, not two: 

```cpp
static const char * const requested_attributes[] =
{
  "printer-state",
  "printer-state-message",
  "printer-state-reasons"
};

ippAddStrings(request, IPP_TAG_OPERATION, IPP_TAG_KEYWORD,
              "requested-attributes", 3, NULL,
              requested_attributes);
 ```

Interestingly, the [test classes use sizeof/size](https://github.com/apple/cups/blob/82e3ee0e3230287b76a76fb8f16b92ca6e50b444/cups/testclient.c#L709) computations instead.

Does JNA do the String[] -> char* conversion for us?

Reply:
You missed an item 


Where:
* `ipp_t *ipp` = `request` (a Pointer object)
* `ipp_tag_t group` = `IPP_TAG_SUBSCRIPTION`
* `ipp_tag_t value_tag` = **`IPP.TAG_KEYWORD`**
* `name` = :no_entry_sign: "notify-events"
* `int num_values` = `3` :warning: should be `2`
* `char *language` = `CHARSET`
* `char *values` = `{"job-state-changed", "printer-state-changed"}`

Reply:
Fixed via 7f8c68c.